Documentation/coreos.rst

```diff
@@ -0,0 +1,376 @@
+        command:
+        - /hyperkube
+        - controller-manager
+        - --allocate-node-cidrs
+        - --cluster-cidr=${CLUSTER_CIDR}
+        - --service-cluster-ip-range=${SERVICE_CLUSTER_IP_RANGE}
+        - --node-cidr-mask-size=${NODE_CIDR_MASK_SIZE}
+        - --master=http://127.0.0.1:8080
+        - --leader-elect=true
+        - --service-account-private-key-file=/etc/kubernetes/ssl/apiserver-key.pem
+        - --root-ca-file=/etc/kubernetes/ssl/ca.pem
+        resources:
+          requests:
+            cpu: 200m
+        livenessProbe:
+          httpGet:
+            host: 127.0.0.1
+            path: /healthz
+            port: 10252
+          initialDelaySeconds: 15
+          timeoutSeconds: 15
+        volumeMounts:
+        - mountPath: /etc/kubernetes/ssl
+          name: ssl-certs-kubernetes
+          readOnly: true
+        - mountPath: /etc/ssl/certs
+          name: ssl-certs-host
+          readOnly: true
+      volumes:
+      - hostPath:
+          path: /etc/kubernetes/ssl
+        name: ssl-certs-kubernetes
+      - hostPath:
+          path: /usr/share/ca-certificates
+        name: ssl-certs-host
+
+
+Step 5: Setup kube-scheduler on the master nodes
+----------------------------------------------------
+Cilium has no special requirements for setting up the kube-scheduler on master nodes. Follow the `CoreOS instructions for setting up kube-scheduler <https://coreos.com/kubernetes/docs/latest/deploy-master.html#set-up-the-kube-scheduler-pod>`_.
+
+Make sure that you set the version of the kube-apiserver to whatever version you are using for the Kubelet as well, e.g.:
+
+::
+
+    quay.io/coreos/hyperkube:v${KUBE_VERSION}_coreos.0
+
+Step 6: Start Services on Nodes
+--------------------------------
+
+Start kubelet on all nodes:
+
+::
+
+    sudo systemctl start kubelet
+
+To have kubelet start after a reboot, run:
+
+::
+
+    sudo systemctl enable kubelet
+
+Step 7: Health Check of Kubernetes Services
+---------------------------------------------
+
+Follow `the CoreOS instructions to health check Kubernetes services <https://coreos.com/kubernetes/docs/latest/deploy-master.html#basic-health-checks>`_.
+
+
+Step 8: Setup Kubectl to Communicate With Your Cluster
+---------------------------------------------------------
+
+
+Follow `the CoreOS instructions to download kubectl <https://coreos.com/kubernetes/docs/latest/configure-kubectl.html#download-the-kubectl-executable>`_.
+
+* Replace ${MASTER_HOST} with the master node address or name used in previous steps
+* Replace ${CA_CERT} with the absolute path to the ca.pem created in previous steps
+* Replace ${ADMIN_KEY} with the absolute path to the admin-key.pem created in previous steps
+* Replace ${ADMIN_CERT} with the absolute path to the admin.pem created in previous steps
+
+:: 
+
+    kubectl config set-cluster default-cluster --server=https://${MASTER_IP} --certificate-authority=${CA_CERT} --embed-certs=true 
```

Comment:
So that we have a kubeconfig dedicated for cilium without copying (possible) sensitive data from the default kubeconfig where users store all kubernetes clusters.